docs/setup.asciidoc

```diff
@@ -34,19 +41,104 @@ https://www.nuget.org/packages/Elastic.Apm.SqlClient[**Elastic.Apm.SqlClient**]:
 
 This package contains https://www.nuget.org/packages/System.Data.SqlClient[System.Data.SqlClient] and https://www.nuget.org/packages/Microsoft.Data.SqlClient[Microsoft.Data.SqlClient] monitoring related code.
 
+
 [float]
 === Quick start
 
+* <<setup-dotnet-net-core>>
 * <<setup-asp-net-core>>
 * <<setup-asp-net>>
 * <<setup-ef6>>
 * <<setup-sqlclient>>
 * <<setup-general>>
 
+
+[float]
+[[setup-dotnet-net-core]]
+==== NET Core
+
+On .NET Core the agent can be registered on the `IHostBuilder`. This applies to both ASP.NET Core and to other .NET Core applications that depend on `IHostBuilder` like https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services[background tasks]. In this case you need to reference the `Elastic.Apm.NetCoreAll` package.
+
+
+[source,csharp]
+----
+using Elastic.Apm.NetCoreAll;
+
+namespace MyApplication
+{
+  public class Program
+  {
+    public static IHostBuilder CreateHostBuilder(string[] args) =>
+        Host.CreateDefaultBuilder(args)
+            .ConfigureWebHostDefaults(webBuilder => { webBuilder.UseStartup<Startup>(); })
+            .UseAllElasticApm();
+
+    public static void Main(string[] args) => CreateHostBuilder(args).Build().Run();
+  }
+}
+----
+
+With the `UseAllElasticApm()` the agent with all its components is turned on. On ASP.NET Core it'll automatically capture incoming requests, database calls through supported technologies, outgoing HTTP requests and so on.
+
+Additionally the `UseAllElasticApm` will also add an `ITracer` to the Dependency Injection system. This means you can use the <<public-api>> in your code like this:
+
+[source,csharp]
+----
+using Elastic.Apm.Api;
+
+namespace WebApplication.Controllers
+{
+    public class HomeController : Controller
+    {
+        private readonly ITracer _tracer;
+        public HomeController(ITracer tracer) //inject ITracer
+        => _tracer = tracer;
+
+        public IActionResult Index()
+        {
+            //use ITracer
+            var span = _tracer.CurrentTransaction?.StartSpan("MySampleSpan", "Sample");
+            try
+            {
+                //your code here
+            }
+            catch (Exception e)
+            {
+                span?.CaptureException(e);
+                throw;
+            }
+            finally { }
+            {
+                span?.End();
+            }
+            return View();
+        }
+    }
+}
+----
+
+Similarly to this ASP.NET Core controller, you can use the same approach with `IHostedService` implementations.
+
+The `Elastic.Apm.NetCoreAll` will reference every agent components. This is usually not a problem. Nevertheless in case you want to keep dependencies minimal you can also reference the `Elastic.Apm.Extensions.Hosting` and use the `UseElasticApm` method instead of `UseAllElasticApm`. With this you can control what the agent will listen for.
```

Comment:
Related to the comment above: here we go even further and discuss how to control specific instrumentation modules. I feel this must be documented, but not sure this is the good place for it.